app/controllers/login_controller.rb

```diff
@@ -0,0 +1,13 @@
+class LoginController < ApplicationController
+
+  def create
+    @market = Market.find(params[:id])
+    redirect_to market_path(@market.id)
+  end
+
```

Comment:
I really like that you included a controller to manage your login. Instead of using the create route here for logging in the market, I think it makes sense to use a custom route. The create route by rails convention is usually used to create a new record in the database, and since you are using it here for a different purpose it makes sense to give it a different name.